bitnami/moodle/templates/svc.yaml

```diff
@@ -41,4 +41,14 @@ spec:
       {{- else if eq .Values.service.type "ClusterIP" }}
       nodePort: null
       {{- end }}
+      {{- if .Values.metrics.enabled }}
+    - name: metrics
+      port: {{ .Values.service.metricsPort }}
+      targetPort: metrics
+      {{- if (and (or (eq .Values.service.type "NodePort") (eq .Values.service.type "LoadBalancer")) (not (empty .Values.service.nodePorts.metrics))) }}
+      nodePort: {{ .Values.service.nodePorts.metrics }}
+      {{- else if eq .Values.service.type "ClusterIP" }}
+      nodePort: null
+      {{- end }}
+      {{- end }}
```

Comment:
Would it make sense to create a different service for exposing the metrics?
I mean, let's say that I use a "LoadBalancer" service type because I want to expose my Moodle website. If I use this approach, I will be exposing the metrics to the outside world to... However, if there are two different services, I can expose the http/https ports while keeping the metrics private.

Reply:
Hi @juan131, you are right, the better aproch is creating a sidecar, but in this case is not posible because this chart do not support the extraPorts 
in order to support the extraport we need to add the following code
**svc.yaml**
`{ { - if .Values.service.extraPorts } }
{ { - include "moodle.tplValue" (dict "value" .Values.service.extraPorts "context" $) | nindent 4 } }
{ { - end } }`

**_helper.tpl**
`{{/*
Renders a value that contains template.
Usage:
{{ include "wordpress.tplValue" ( dict "value" .Values.path.to.the.Value "context" $) }}
*/}}
{{- define "moodle.tplValue" -}}
    {{- if typeIs "string" .value }}
        {{- tpl .value .context }}
    {{- else }}
        {{- tpl (.value | toYaml) .context }}
    {{- end }}
{{- end -}}
`

Reply:
I ended up doing what @juan131 suggested and created a separated service for exposing the metrics; `metrics-svc.yaml`